### PR TITLE
Fixes to make sota-tools compatible with gcc v4.x (requirement from AGL/jsmoeller) and the respective test

### PR DESCRIPTION
**Pull request by OYTIS**
Causes for incompatibilities

1. std::stringstream is not moveable in gcc v4.x
2. Enums are not namespaces (I've also encountered this error on newer versions of GCC).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@ env:
   - DOCKERFILE=Dockerfile.nop11 CACHE=latest SCRIPT=src/scripts/test.sh
   - DOCKERFILE=Dockerfile.deb-stable CACHE=latest-deb-stable SCRIPT=src/scripts/test.sh DARGS="-eBUILD_ONLY=1 -eBUILD_DEB=1"
   - DOCKERFILE=Dockerfile.deb-unstable CACHE=latest-deb-unstable SCRIPT=src/scripts/test.sh DARGS="-eBUILD_ONLY=1"
+  - DOCKERFILE=Dockerfile.gcc4 CACHE=latest SCRIPT=src/scripts/test.sh DARGS="-eBUILD_ONLY=1 -eNO_FORMAT_CHECK=1"
```
Comment (cajun-rat) [inline .travis.yml]: +1

File: CMakeLists.txt [hunk 2]
```diff
@@ -25,7 +25,7 @@ set(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake-modules)
 # find all required libraries
 set(Boost_USE_STATIC_LIBS ON)
 find_package(PkgConfig REQUIRED)
-find_package(Boost COMPONENTS system filesystem thread program_options log_setup log regex chrono random REQUIRED)
+find_package(Boost 1.57.0 COMPONENTS system filesystem thread program_options log_setup log regex chrono random REQUIRED)
```
Comment (cajun-rat) [inline CMakeLists.txt]: Is this a minimum version number?

Comment (cajun-rat) [reply CMakeLists.txt]: Sorry, dumb question. I just RTFM and this is a minimum.

File: Dockerfile.gcc4 [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# It is important to run these in the same RUN command, because otherwise
+# Docker layer caching breaks us
+RUN apt-get update && apt-get -y install wget liblzma-dev bison e2fslibs-dev libgpgme11-dev libglib2.0-dev gcc-4.9 g++-4.9 make cmake git psmisc dbus python3-dbus python3-gi python3-openssl libdbus-1-dev libjansson-dev libgtest-dev libssl-dev autoconf automake pkg-config libtool libexpat1-dev libboost-program-options-dev libboost-test-dev libboost-random-dev libboost-regex-dev libboost-dev libboost-system-dev libboost-date-time-dev libboost-thread-dev libboost-log-dev libjsoncpp-dev curl libcurl4-openssl-dev lcov clang clang-format-3.8
+RUN apt-get update && apt-get -y install ostree libostree-dev libsodium-dev libarchive-dev python3-venv python3-dev valgrind libp11-dev softhsm2 opensc libengine-pkcs11-openssl
+RUN rm /usr/bin/gcc /usr/bin/gcc-ar /usr/bin/gcc-nm /usr/bin/gcc-ranlib /usr/bin/g++ && \
+    ln -s gcc-4.9 /usr/bin/gcc && \
+    ln -s gcc-ar-4.9 /usr/bin/gcc-ar && \
+    ln -s gcc-nm-4.9 /usr/bin/gcc-nm && \
+    ln -s gcc-ranlib-4.9 /usr/bin/gcc-ranlib && \
+    ln -s g++-nm-4.9 /usr/bin/g++
```
Comment (patriotyk) [inline Dockerfile.gcc4]: In debian there is "update-alternatives" command that do such symlinks and that symlinks will be persistent after package updates

Comment (OYTIS) [reply Dockerfile.gcc4]: Thank you, I'll take a look at it.

**Pull request merged**